### PR TITLE
Fix/bad data continue

### PR DESCRIPTION
**Pull request by ADavidson02**
## Is this a fix or a feature?
Feature 

## What is the change?
The posting of data now has conditional added on

## Where should the reviewer start?
fetch.js line 30

## How should this be tested?
Try and press the submit buttons on any of the field without first entering any data.
An alert should pop up informing the user of the missing data. Once all fields are filled the post request will go through.

## Relevant Ticket(s)

@Shawntru 
@adavidson02
@sarah01123

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -69,6 +71,10 @@ <h2 class="sidebar-header-name" id="sidebarName">Name Name</h2>
       </section>
       <section class="main-column-activity">
         <section class="infoContainer-cardContainer-card-horizontal">
+          <label for="todays-steps"><input type="text" class="new-activity-input-steps" placeholder="Number of steps" onkeyup="value=value.replace(/[^\d]/g,'')">
```
Comment (saraho1123) [inline src/index.html:74]: This is really cool! I definitely want you to teach me about this!

File: src/scripts.js [hunk 2]
```diff
@@ -97,6 +115,145 @@ function getUserById(id, listRepo) {
...
+function getTodaysDate() {
+  let today = new Date();
+  let dd = String(today.getDate()).padStart(2, '0');
+  let mm = String(today.getMonth() + 1).padStart(2, '0'); //January is 0!
+  let yyyy = today.getFullYear();
+  return today = yyyy + '/' + mm + '/' + dd;
+}
+
+function submitNewHydration(userNowId) {
+  // let todaysDate = getTodaysDate();
```
Comment (saraho1123) [inline src/scripts.js:139]: I know this is code that I contributed to. It is working with getTodaysDate in the PR I am about to submit.

File: src/scripts.js [hunk 3]
```diff
@@ -97,6 +115,145 @@ function getUserById(id, listRepo) {
...
+  let todaysDate = '2020/9/31';
+  /*
```
Comment (saraho1123) [inline src/scripts.js:141]: I am about to submit a PR that has all these comments cleaned up for all the functions in script that have to do with users being able to submit data (POSTing).

File: src/scripts.js (hunk 2 above)
Comment (ADavidson02) [reply src/scripts.js:139]: Ok I was not sure which route we were taking to reload all of the updated information on the page so I did not touch any thing besides adding the conditionals.

File: src/scripts.js (hunk 3 above)
Comment (ADavidson02) [reply src/scripts.js:141]: Sounds perfect, I may have to have you explain this.

File: src/scripts.js [hunk 4]
```diff
@@ -97,6 +115,145 @@ function getUserById(id, listRepo) {
...
+
+function submitNewActivity(userNowId) {
+  // let todaysDate = getTodaysDate();
+  let todaysDate = '2020/9/31';
+  let newStepsActivityData = newStepsInput.value;
+  let newMinutesActivityData = newStairsInput.value;
+  let newStairsActivityData = newActiveMinutesInput.value;
+  if (!newStepsActivityData) {
+    return alert('Please enter a number of steps');
+  } else if (newStepsActivityData > 10000) {
```
Comment (saraho1123) [inline src/scripts.js:177]: Thanks for adding all these conditionals Amanda! I think I made the suggestion of limiting to 10,000 steps, but maybe we should amend that. I know I have done some activities that put me WAY over that number (I think my max ever was somewhere around 32,000). This definitely isn't a big deal though! The code logic is sound, and that is what matters! :)

File: src/index.html (hunk 1 above)
Comment (ADavidson02) [reply src/index.html:74]: The label is necessary for our lighthouse score.

File: src/scripts.js [hunk 5]
```diff
@@ -127,12 +284,48 @@ function makeRandomDate(userStorage, id, dataSet) {
 }
 
 function addHydrationInfo(id, hydrationInfo, dateString, userStorage, laterDateString) {
-  hydrationToday.insertAdjacentHTML('afterBegin', `<p>You drank</p><p><span class="number">${hydrationInfo.calculateDailyOunces(id, dateString)}</span></p><p>oz water today.</p>`);
+  // hydrationToday.innerHTML = '';
```
Comment (saraho1123) [inline src/scripts.js:287]: As with all the comments (and console.logs) in scripts right now, I do have those all cleaned up for the next PR!

**Review by saraho1123 — APPROVED** [review]
Thanks for integrating all this Amanda! I made a couple comments directly on the code, but any of the commented out notes and code, I have cleaned up for the next PR! No need to worry about that stuff for this one.

File: src/index.html (hunk 1 above)
Comment (Shawntru) [reply src/index.html:74]: Great, thank you for putting in those labels

**Review by Shawntru — APPROVED** [review]
Approved

**Pull request merged**